packages/keystone/plugins/searchBy.js

```diff
@@ -0,0 +1,189 @@
+const { isEmpty, isString, isNil, isFunction, get, isNumber } = require('lodash')
+
+const { plugin } = require('./utils/typing')
+const { composeResolveInputHook } = require('./utils')
+const { getByCondition, getSchemaCtx } = require('../schema')
+
+
+const SEARCH_FIELD_DEFAULT_VALUE = '""'
+const FIELD_NAME_REGEX = /^[a-z]+$/i
+
+const SEARCH_OPTIONS = {
+    schemaDoc: 'Service field for searching by different fields',
+    type: 'Text',
+    kmigratorOptions: { null: false },
+    defaultValue: SEARCH_FIELD_DEFAULT_VALUE,
+    access: {
+        create: false,
+        read: true,
+        update: true,
+    },
+    knexOptions: { defaultTo: () => SEARCH_FIELD_DEFAULT_VALUE },
+}
+const SEARCH_VERSION_OPTIONS = {
```

Comment:
how is it supposed to be used ?

Reply:
The "searchVersion" field is used as a trigger.
This allows you to control the values of the "search" field from the main model. Linked models can only report that they have updated without interacting with the "search" field directly.

Tracked field updated --> update "searchVersion" field --> need to update "search" field.

Reply:
But it is better to use not a number, but some random value. For example, uuid